lib/metadata/wrapper.js

```diff
@@ -117,9 +117,9 @@ const metadata = {
         });
     },
 
-    listObject: (bucketName, prefix, marker, delimiter, maxKeys, log, cb) => {
+    listObject: (bucketName, listingParams, log, cb) => {
```

Comment:
For the ability to extend. Listing does not always take only these arguments, for example listing all versions of a bucket uses `max-keys`, `key-marker`, and `version-id-marker`.